docs: add specific docs for each command and args (#58)
rook/kubectl-rook-ceph: docs/debug.md

2. Start a new debug deployment where operations can be performed directly against the mon or OSD without that daemon running
   a. The main container sleeps so you can connect and run the ceph commands
   b. Liveness and startup probes are removed
   c. If alternate Image is passed by --alternate-image flag then the new debug deployment container will be using alternate Image.

Debug mode provides these options:

1. [Start](#start-debug-mode) the debug deployment for troubleshooting.
2. [Stop](#stop-debug-mode) the temporary debug deployment
3. Update the resource limits for the deployment pod [advanced option](#advanced-options).

## Start debug mode

In this example we are using `mon-b` deployment

```bash
kubectl rook-ceph debug start rook-ceph-mon-b

# setting debug mode for "rook-ceph-mon-b"
# setting debug command to main container
# deployment.apps/rook-ceph-mon-b scaled
# deployment.apps/rook-ceph-mon-b-debug created
```

Now connect to the daemon pod and perform operations:

```console
kubectl exec <debug-pod> -- <ceph command>
```

When finished, stop debug mode and restore the original daemon by running the command in the next section.

> ### Stop debug mode

Author: done

Stop the deployment `mon-b` that is started above example.

```bash
kubectl rook-ceph debug stop rook-ceph-mon-b

# setting debug mode for "rook-ceph-mon-b-debug"
# removing debug mode from "rook-ceph-mon-b-debug"
# deployment.apps "rook-ceph-mon-b-debug" deleted
# deployment.apps/rook-ceph-mon-b scaled
```

> ### Advanced Options

Reviewer: ```suggestion
## Advanced Options
```

Author: ack

# Debug Mode

Debug mode can be useful when a mon or OSD needs advanced maintenance operations that require the daemon to be stopped. Ceph tools such as `ceph-objectstore-tool`,`ceph-bluestore-tool`, or `ceph-monstore-tool` are commonly used in these scenarios. Debug mode will set up the mon or OSD so that these commands can be run.

Debug mode will automate the following:

1. Scale down the existing mon or OSD deployment
2. Start a new debug deployment where operations can be performed directly against the mon or OSD without that daemon running
   a. The main container sleeps so you can connect and run the ceph commands
   b. Liveness and startup probes are removed
   c. If alternate Image is passed by --alternate-image flag then the new debug deployment container will be using alternate Image.

> ## Usage

Reviewer: I think no need for this header
```suggestion
```

Author: removed

> Debug mode provide these options:

Reviewer: ```suggestion
Debug mode provides these options:
```

Author: ack

> 1. To [start](#start-debug-mode) the deployment for troubleshooting.

Reviewer: ```suggestion
1. [Start](#start-debug-mode) the debug deployment for troubleshooting.
```

Author: ack

> 2. To [stop](#stop-debug-mode) the deployment that is started.

Reviewer: ```suggestion
2. [Stop](#stop-debug-mode) the temporary debug deployment
```

Author: ack

> 3. To update the resource limits for deployment pod [advanced option](#advanced-options).

Reviewer: ```suggestion
3. Update the resource limits for the deployment pod [advanced option](#advanced-options).
```

Author: done

> ### Start debug mode

Reviewer: Start/Stop headers can be up a level
```suggestion
## Start debug mode
```

Author: changed

In this example we are using `mon-b` deployment

```bash
kubectl rook-ceph debug start rook-ceph-mon-b

# setting debug mode for "rook-ceph-mon-b"
# setting debug command to main container
# deployment.apps/rook-ceph-mon-b scaled
# deployment.apps/rook-ceph-mon-b-debug created
```

Now connect to the daemon pod and perform operations:

```console
kubectl exec <debug-pod> -- <ceph command>
```

> When finished, stop debug mode and restore the original daemon by running command [below](#stop-debug-mode).

Reviewer: ```suggestion
When finished, stop debug mode and restore the original daemon by running the command in the next section.
```

Author: changed

# Debug Mode

Debug mode can be useful when a mon or OSD needs advanced maintenance operations that require the daemon to be stopped. Ceph tools such as `ceph-objectstore-tool`,`ceph-bluestore-tool`, or `ceph-monstore-tool` are commonly used in these scenarios. Debug mode will set up the mon or OSD so that these commands can be run.

Debug mode will automate the following:

1. Scale down the existing mon or OSD deployment

> 2. Start a new debug deployment where operations can be performed directly against the mon or OSD without that daemon running
> 
>    a. The main container sleeps so you can connect and run the ceph commands
> 
>    b. Liveness and startup probes are removed
> 
>    c. If alternate Image is passed by --alternate-image flag then the new debug deployment container will be using alternate Image.

Reviewer 2: ?

```suggestion
2. Start a new debug deployment where operations can be performed directly against the mon or OSD without that daemon running
   a. The main container sleeps so you can connect and run the ceph commands
   b. Liveness and startup probes are removed
   c. If alternate Image is passed by --alternate-image flag then the new debug deployment container will be using alternate Image.
```

Author: done
